packages/vaadin-virtual-list/src/vaadin-virtual-list.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { PolymerElement, html } from '@polymer/polymer/polymer-element.js';
+import { ThemableMixin } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
+import { ElementMixin } from '@vaadin/vaadin-element-mixin/vaadin-element-mixin.js';
+import { Virtualizer } from './virtualizer.js';
+
+/**
+ * `<vaadin-virtual-list>` is a Web Component for displaying a virtual/infinite list or items.
+ *
+ * ```html
+ * <vaadin-virtual-list></vaadin-virtual-list>
+ * ```
+ *
+ * ```js
+ * const list = document.querySelector('vaadin-virtual-list');
+ * list.items = items; // An array of data items
+ * list.renderer = (root, list, {item, index}) => {
+ *   root.textContent = `#${index}: ${item.name}`
+ * }
+ * ```
+ *
+ * See [Virtual List](https://vaadin.com/docs/latest/ds/components/virtual-list) documentation.
+ *
+ * @extends HTMLElement
+ * @mixes ElementMixin
+ * @mixes ThemableMixin
+ */
+class VirtualListElement extends ElementMixin(ThemableMixin(PolymerElement)) {
+  static get template() {
+    return html`
+      <style>
+        :host {
+          display: block;
+          height: 200px;
```

Comment:
Added a default value to have `<vaadin-virtual-list>` with no explicitly defined height to not appear invisible.